code/modules/mob/living/carbon/xenomorph/castes/pyrogen/abilities_pyrogen.dm

```diff
@@ -0,0 +1,529 @@
+	target_flags = ABILITY_TURF_TARGET
+	ability_cost = 50
+	cooldown_duration = 8 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_FIRENADO,
+	)
+
+/obj/effect/xenomorph/firenado
+	name = "Plasma Whirlwind"
+	desc = "A glowing whirlwind of... cold plasma? seems to \"burn\" "
+	icon = 'icons/effects/64x64.dmi'
+	icon_state = "whirlwind"
+	anchored = TRUE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	pixel_x = -16
+	/// How many process ticks can we do before we are deleted/we extinguish
+	var/burn_ticks_left = 20
+	/// Target turf to bias going towards
+	var/turf/target
+
+/obj/effect/xenomorph/firenado/New(loc, arg_target)
+	. = ..()
+	target = arg_target
+
+/obj/effect/xenomorph/firenado/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+	var/static/list/connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_cross),
+	)
+	AddElement(/datum/element/connect_loc, connections)
+
+
+/obj/effect/xenomorph/firenado/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/// mob affecting
+/obj/effect/xenomorph/firenado/proc/mob_act(mob/living/carbon/human/target)
+	var/damage = PYROGEN_TORNADE_HIT_DAMAGE
+	if(target.has_status_effect(STATUS_EFFECT_MELTING_FIRE))
+		var/datum/status_effect/stacking/melting_fire/debuff = target.has_status_effect(STATUS_EFFECT_MELTING_FIRE)
+		debuff.add_stacks(PYROGEN_TORNADO_MELTING_FIRE_STACKS)
+	target.apply_status_effect(STATUS_EFFECT_MELTING_FIRE, PYROGEN_TORNADO_MELTING_FIRE_STACKS)
+	target.take_overall_damage(damage, BURN, FIRE)
+
+///Effects applied to a mob that crosses a burning turf
+/obj/effect/xenomorph/firenado/proc/on_cross(datum/source, mob/living/carbon/human/target, oldloc, oldlocs)
+	if(!istype(target))
+		return
+	mob_act(target)
+	qdel(src)
+
+/obj/effect/xenomorph/firenado/Bump(atom/target)
+	. = ..()
+	if(iswallturf(target))
+		if(!istype(target,/turf/closed/wall/resin))
+			var/turf/closed/wall/wall = target
+			wall.take_damage(PYROGEN_TORNADE_HIT_DAMAGE, BURN)
+			qdel(src)
+			return
+	else if(isobj(target) && !istype(target, type))
+		var/obj/object = target
+		object.take_damage(PYROGEN_TORNADE_HIT_DAMAGE, BURN)
+		qdel(src)
+		return
+
+/obj/effect/xenomorph/firenado/process()
+	var/turf/current_location = loc
+	if(!istype(current_location)) //Is it a valid turf?
+		qdel(src)
+		return
+	for(var/mob/living/carbon/human/humie in current_location)
+		if(humie.stat == DEAD)
+			continue
+		mob_act(humie)
+		burn_ticks_left = 0
+	var/list/move_dirs = list(NORTH, SOUTH, EAST, WEST)
+	if(target)
+		var/target_dir = get_dir(src, target)
+		// double bias towards target dir
+		move_dirs.Add(target_dir)
+		move_dirs.Add(target_dir)
+
+	// before moving so that if we hit someone , we don't also put down a fire that they will instantly gain another stack from
+	if(!locate(/obj/melting_fire) in current_location)
+		new /obj/melting_fire(current_location)
+	step(src,pick(move_dirs))
+	burn_ticks_left--
+	if(burn_ticks_left <= 0)
```

Comment:
why not use QDEL_IN_STOPPABLE for the lifetime?

Reply:
process() may lag/vary on time.

Reply:
I dont understand

Reply:
This keeps the damage consistent , if i was using QDEL_IN_STOPPABLE then the effectiveness would be massively reduced as timers have max priority but object processing doesn't for the MC.   The MC would always process timers but sometimes skip obj processing if the CPU usage is too high.

Reply:
That is not a good reason for doing extra processing here, change it